File: app/src/main/java/wangdaye/com/geometricweather/db/converters/WeatherCodeConverter.java
package wangdaye.com.geometricweather.db.converters;

import androidx.room.TypeConverter;

import wangdaye.com.geometricweather.common.basic.models.weather.WeatherCode;

public class WeatherCodeConverter {

    @TypeConverter
    public WeatherCode convertToEntityProperty(String databaseValue) {
        if (databaseValue == null) {
            return null;
        }
        // use get instance method but not getValue method.
        return WeatherCode.getInstance(databaseValue);
    }

    @TypeConverter
    public String convertToDatabaseValue(WeatherCode entityProperty) {
        if (entityProperty == null) {
            return null;
        }
        return entityProperty.getId();
    }
}

File: app/src/main/java/wangdaye/com/geometricweather/db/converters/WeatherCodeConverter.kt
package wangdaye.com.geometricweather.db.converters

import androidx.room.TypeConverter
import wangdaye.com.geometricweather.common.basic.models.weather.WeatherCode

class WeatherCodeConverter {

    @TypeConverter
    fun convertToEntityProperty(databaseValue: String?): WeatherCode? {
        if (databaseValue == null) {
            return null
        }
        // use get instance method but not getValue method.
        return WeatherCode.getInstance(databaseValue)
    }

    @TypeConverter
    fun convertToDatabaseValue(entityProperty: WeatherCode?): String? {
        return entityProperty?.id
    }
}
